sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -559,6 +561,17 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
         })
         Join(newLeft, newRight, joinType, newCondition, hint)
 
+      case _: Union =>
+        var first = true
+        plan.mapChildren { child =>
+          if (first) {
+            first = false
```

Comment:
can we add some comments to explain why union's first child is special?

Reply:
Sure, added in: https://github.com/apache/spark/pull/37334/commits/db10b5f698b45fe849133e8f347a27abb3104ec8